AdaptorsFunctionality/src/au/gov/asd/tac/constellation/functionality/adaptors/dataaccess/plugins/utilities/GDELTImportingUtilities.java

```diff
@@ -0,0 +1,480 @@
+package au.gov.asd.tac.constellation.functionality.adaptors.dataaccess.plugins.utilities;
+
+import au.gov.asd.tac.constellation.graph.processing.GraphRecordStore;
+import au.gov.asd.tac.constellation.graph.processing.GraphRecordStoreUtilities;
+import au.gov.asd.tac.constellation.graph.processing.RecordStore;
+import au.gov.asd.tac.constellation.graph.schema.analytic.concept.AnalyticConcept;
+import au.gov.asd.tac.constellation.graph.schema.analytic.concept.SpatialConcept;
+import au.gov.asd.tac.constellation.graph.schema.analytic.concept.TemporalConcept;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.LocalDate;
+import java.time.temporal.ChronoField;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/*
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ *
+ * @author canis_majoris
+ */
+public class GDELTImportingUtilities {
+    
+    private static final String HEADER = "http://data.gdeltproject.org/gkg/";
+    private static final String FOOTER = ".gkg.csv";
+    private static final String ZIPPER = ".zip";
+    
+    public static RecordStore retrieveEntities(LocalDate localDate, List<String> options, int limit) throws MalformedURLException, IOException {
+        
+        final int h = localDate.get(ChronoField.YEAR);
+        final int m = localDate.get(ChronoField.MONTH_OF_YEAR);
+        final int d = localDate.get(ChronoField.DAY_OF_MONTH);
+        
+        final String day = String.format("%04d%02d%02d", h, m, d);

```

Comment:
Surely you could use a convenience function (perhaps alongside a DateTimeFormatter) to do this?